src/System.Security.Cryptography.Csp/src/System/Security/Cryptography/CapiHelper.DSA.cs

```diff
@@ -0,0 +1,399 @@
+                    //  BYTE[lenX]      X (if private)
+
+                    bw.Write((int)(isPrivate ? DSS_PRIV_MAGIC_VER3 : DSS_PUB_MAGIC_VER3));
+                    bw.Write((uint)(bitLenP));
+                    bw.Write((uint)(dsaParameters.Q.Length * 8));
+                    bw.Write((uint)(bitLenJ));
+
+                    if (isPrivate)
+                    {
+                        bw.Write((uint)dsaParameters.X.Length * 8);
+                    }
+
+                    WriteDSSSeed(dsaParameters, bw);
+
+                    bw.WriteReversed(dsaParameters.P);
+                    bw.WriteReversed(dsaParameters.Q);
+                    bw.WriteReversed(dsaParameters.G);
+
+                    if (bitLenJ != 0)
+                    {
+                        bw.WriteReversed(dsaParameters.J);
+                    }
+
+                    bw.WriteReversed(dsaParameters.Y);
+
+                    if (isPrivate)
+                    {
+                        bw.WriteReversed(dsaParameters.X);
+                    }
+                }
+                else
+                {
+                    // We need to build a key blob as follows:
+                    //  DWORD           magic (DSSPUBKEY)
+                    //  DWORD           bitlen (DSSPUBKEY)
+                    //  BYTE[len]       P
+                    //  BYTE[DSS_Q_LEN] Q
+                    //  BYTE[len]       G
+                    //  BYTE[20]        X (if private)
+                    //  BYTE[len]       Y (if public)
+                    //  DWORD           counter (DSSSEED)
+                    //  BYTE[20]        seed (DSSSEED)
+
+                    bw.Write((int)(isPrivate ? DSS_PRIVATE_MAGIC : DSS_MAGIC));
+                    bw.Write((uint)(bitLenP));
+                    bw.WriteReversed(dsaParameters.P);
+                    bw.WriteReversed(dsaParameters.Q);
+                    bw.WriteReversed(dsaParameters.G);
+
+                    if (isPrivate)
+                    {
+                        bw.WriteReversed(dsaParameters.X);
+                    }
+                    else
+                    {
+                        bw.WriteReversed(dsaParameters.Y);
+                    }
+
+                    WriteDSSSeed(dsaParameters, bw);
+                }
+
+                bw.Flush();
+                byte[] key = ms.ToArray();
+                return key;
+            }
+        }
+
+        /// <summary>
+        /// Helper for DSACryptoServiceProvider.ExportParameters()
+        /// </summary>
+        internal static DSAParameters ToDSAParameters(this byte[] cspBlob, bool includePrivateParameters, SafeKeyHandle safeKeyHandle)
+        {
+            try
+            {
+                using (var ms = new MemoryStream(cspBlob))
+                using (var br = new BinaryReader(ms))
+                {
+                    byte bType;
+                    byte bVersion;
+                    ReadKeyBlobHeader(br, out bType, out bVersion);
```

Comment:
I will remove the out parameter. The variable was useful for debugging sanity check, and I also had an Assert at one point which I removed for concern over compatibility. It is not verified or used otherwise for compatibility (as done on RSA side as well). Here's the comment on the RSA side:
// BLOBHEADER.bType: Expected to be 0x6 (PUBLICKEYBLOB) or 0x7 (PRIVATEKEYBLOB), though there's no check for backward compat reasons.